test/helper/resource.go

```diff
@@ -152,6 +152,32 @@ func NewService(namespace string, name string) *corev1.Service {
 	}
 }
 
+// NewNodePortService will build a nodeport service object.
+func NewNodePortService(namespace string, name string) *corev1.Service {
```

Comment:
We can add a new input parameter to the `NewService` method to indicate the service type.